framework/encryption/__init__.py

```diff
@@ -6,13 +6,22 @@
 
 SENSITIVE_DATA_KEY = jwe.kdf(settings.SENSITIVE_DATA_SECRET.encode('utf-8'), settings.SENSITIVE_DATA_SALT.encode('utf-8'))
 
+def ensure_str(value):
```

Comment:
Just for clarity you may want to name this `ensure_bytes`.
In python 2.7 `str == bytes` but that is often confused with `unicode`.
It's fairly difficult to determine what a string literal will end up being.

```python
isinstance(b'example text', bytes)  # True
isinstance(b'example text', str) # True
isinstance(b'example text', unicode) # False
```

```python
isinstance('example text', bytes)  # True
isinstance('example text', str) # True
isinstance('example text', unicode) # False
```

```python
isinstance(u'example text', bytes)  # False
isinstance(u'example text', str) # False
isinstance(u'example text', unicode) # True
```

```python
from __future__ import unicode_literals
isinstance('example text', bytes)  # False
isinstance('example text', str) # False
isinstance('example text', unicode) # True
```

```python
from __future__ import unicode_literals
isinstance(b'example text', bytes)  # True
isinstance(b'example text', str) # True
isinstance(b'example text', unicode) # False
```

But wait there's more

```python

In [43]: x = u'x'

In [44]: x.encode?
Docstring:
S.encode([encoding[,errors]]) -> string or unicode

Encodes S using the codec registered for encoding. encoding defaults
to the default encoding. errors may be given to set a different error
handling scheme. Default is 'strict' meaning that encoding errors raise
a UnicodeEncodeError. Other possible values are 'ignore', 'replace' and
'xmlcharrefreplace' as well as any other name registered with
codecs.register_error that can handle UnicodeEncodeErrors.
Type:      builtin_function_or_method

In [45]: x = b'x'

In [46]: x.encode?
Docstring:
S.encode([encoding[,errors]]) -> object

Encodes S using the codec registered for encoding. encoding defaults
to the default encoding. errors may be given to set a different error
handling scheme. Default is 'strict' meaning that encoding errors raise
a UnicodeEncodeError. Other possible values are 'ignore', 'replace' and
'xmlcharrefreplace' as well as any other name registered with
codecs.register_error that is able to handle UnicodeEncodeErrors.
Type:      builtin_function_or_method
```

`unicode.encode` -> `bytes`
`bytes.decode` -> `unicode`

I think it would be best to check if the value is already in bytes and `encode` it if not.

Final thought, throw some `assert isinstance`s into the tests to make sure that the value your attempting to send into this function is actually the type you think it is.


PS `# -*- coding: utf-8 -*-` Does weird stuff too. I don't remember what, however. 👍